src/common/server.ts

```diff
@@ -246,15 +246,16 @@ async function createLegacyServer(
   return new LanguageClient(serverId, serverName, serverOptions, clientOptions);
 }
 
-async function showWarningMessageWithLogs(message: string, outputChannel: LogOutputChannel) {
-  const selection = await vscode.window.showWarningMessage(message, "Show Logs");
-  if (selection) {
-    outputChannel.show();
-  }
+function showWarningMessageWithLogs(message: string, outputChannel: LogOutputChannel) {
+  vscode.window.showWarningMessage(message, "Show Logs").then((selection) => {
+    if (selection) {
+      outputChannel.show();
+    }
+  });
```

Comment:
Nit: Maybe add an error handler that logs a message if showing the message box fails for whatever reason
```suggestion
  vscode.window.showWarningMessage(message, "Show Logs").then((selection) => {
    if (selection) {
      outputChannel.show();
    }
  }).catch(ex => logError("Failed to show warning message: {ex}"));
```